contracts/ETO/ETOTokenTerms.sol

```diff
@@ -85,14 +88,19 @@ contract ETOTokenTerms is Math, IContractId {
         EQUITY_TOKEN_NAME = equityTokenName;
         EQUITY_TOKEN_SYMBOL = equityTokenSymbol;
         EQUITY_TOKENS_PER_SHARE = equityTokensPerShare;
+        EQUITY_TOKENS_PRECISION = equityTokensPrecision;
+        EQUITY_TOKENS_POWER = 10**uint256(EQUITY_TOKENS_PRECISION);
+
+        require(equityTokensPerShare >= 0 && equityTokensPerShare % EQUITY_TOKENS_POWER == 0, "NF_SHARES_NOT_WHOLE_TOKENS");
```

Comment:
> equityTokensPerShare >= 0

Why is == 0 allowed ? if not, then 

> equityTokensPerShare % EQUITY_TOKENS_POWER == 0

this only makes sense if ```equityTokensPerShare``` is ULPS, which conflicts with my comment above 

why is any value < EQUITY_TOKENS_POWER allowed? -> For a divisible equity-token to make sense, we need the owners of less than 1 EQUITY-TOKEN to still have a right towards the company, ie. their SHARE needs to be > 0. 
For that to be true

Reply:
yes this is a bug! the intention here was to require TPS > 0 as module division does not catch case of TPS == 0.

there's one more thing we should check: we require in the tests and we assume in many other places that TPS is power of 10. here we can have TPS == 2 tokens and all is fine (contract will work). any clever ideas to check if TPS is a power of ten without logarithm? we could use some loop certainly

Reply:
I think we could just set the way you described it in the other comment:
> EQUITY_TOKENS_PER_SHARE = 10n * 10decimals

so instead of 
```
EQUITY_TOKENS_PER_SHARE = equityTokensPerShare;
```
it would be
```
EQUITY_TOKENS_PER_SHARE = 10*full_tokens_per_share*10**DECIMALS;
```

Reply:
Or we could do this:

```
    function is_power_of_ten(uint x) public pure returns(bool) {
        while (x > 9) {
            if (x % 10 == 0) { x = x / 10; }
            else { return false; }
        }
        return x == 1;
    }
}
```

Reply:
I will go with `10*full_tokens_per_share*10**DECIMALS;`, really good idea!